app/models/event_dispatcher.rb

```diff
@@ -2,19 +2,19 @@
 
 class EventDispatcher < ApplicationRecord
   # Associations
-  belongs_to :job
   belongs_to :event_receiver
+  belongs_to :job
 
   # Validations
   validates :triggered, presence: true
 
-  def dispatch!(job)
+  def dispatch!(user)
     # 1) Don't perform an action if it has already been performed on the rising edge trigger
     # 2) Only perform the dispatch if the event_receiver's trigger condition has been met
     return if triggered || !event_receiver.trigger_condition_met?
 
     # Run each of the event actions corresponding to the job
-    job.event_actions.each(&:run!)
+    job.event_actions.each { |job| job.run!(user) }
```

Comment:
can we call the loop variable `action` instead of `job`?

Reply:
Oops, yeah, that's better